File: Chess/Chess/Blank.java

/**
 * Write a description of class Blank here.
 *
 * @author (your name)
 * @version (a version number or a date)
 */
public class Blank extends Piece
{
    /**
     * Constructor for objects of class Blank
     */
    public Blank(int x, int y)
    {
        super(x,y,0);
        // initialise instance variables
        if((x%2 == 0 && y%2 == 0) || (x%2 != 0 && y%2 != 0)) //white
            this.pieceIcon = 0x25A1;
        else
            this.pieceIcon = 0x25A0;
    }


}

File: Chess/ChessC++/src/Blank.cpp
#include "Blank.h"

Blank::Blank(int x, int y) : Piece(x, y, 0) {
    if ((x % 2 == 0 && y % 2 == 0) || (x % 2 != 0 && y % 2 != 0)) //placement
        pieceIcon = '0'; //0x25A1; // white square
    else
        pieceIcon = '1'; //0x25A0; // black square
}